code/routers/user/reset.js

```diff
@@ -0,0 +1,203 @@
+          'If you did not request this, please ignore this email and your password will remain unchanged.\n'
+      };
+      smtpTransport.sendMail(mailOptions, function(err) {
+        console.log('mail sent');
+        console.log('email: ' + user.email)
+        //req.flash('success', 'An e-mail has been sent to ' + user.email + ' with further instructions.');
+        done(err, 'done');
+      });
+    }
+  ], function(err) {
+    if (err) return next(err);
+    res.redirect('/forgot');
+  });
+});
+
+// :token -> check-token
+router.get('/:token', function(req, res) {
+  console.log('hello')
+  User.findOne({ resetPasswordToken: req.params.token, resetPasswordExpires: { $gt: Date.now() } }, function(err, user) {
+    if (!user) {
+      console.log("Password reset token is invalid or has expired in get method")
+      return res.redirect('/:token');
+    }
+    console.log("returning token: " + req.params.token)
+    res.redirect('token', {token: req.params.token});
+  });
+});
+
+// :token -> reset-password
+router.post('/:token', function(req, res) {
+  async.waterfall([
+    function(done) {
+      User.findOne({ resetPasswordToken: req.params.token, resetPasswordExpires: { $gt: Date.now() } }, function(err, user) {
+        console.log("found user: "+ user)
+
+        if (!user) {
+          console.log("Password reset token is invalid or has expired")
+          return res.redirect('back');
+        }
+        if(req.body.password === req.body.confirm) {
+
+          user.setPassword(req.body.password);
+
+          user.resetPasswordToken = undefined;
+          user.resetPasswordExpires = undefined;
+
+          user.save(function(err) {
+            req.logIn(user, function(err) {
+              done(err, user);
+            });
+          });
+
+        } else {
+            console.log("Passwords do not match")
+            return res.redirect('back');
+        }
+      });
+    },
+    function(user, done) {
+      console.log("sending email")
+      var smtpTransport = nodemailer.createTransport({
+        host: "smtp.gmail.com",
+        auth: {
+          type: "login",
+          user: "DOMAD24901@gmail.com",
+          pass: "DoMADTemp#2020",
+        }
+      });
+      var mailOptions = {
+        to: user.email,
+        from: 'DOMAD24901@gmail.com',
+        subject: 'Your password has been changed',
+        text: 'Hello,\n\n' +
+          'This is a confirmation that the password for your account ' + user.email + ' has just been changed.\n'
+      };
+      smtpTransport.sendMail(mailOptions, function(err) {
+        console.log('Success! Your password has been changed.');
+        done(err, 'done');
+      });
+    }
+  ], function(err) {
+    if (err) return next(err);
+    console.log("No error redirecting to page")
+    res.redirect('/login');
+  });
+});
+
+
+/*
+  contact us api
+
+  sends contact us message to Domad Gmail
+*/
+
+router.get('/contact', function(req, res) {
```

Comment:
same as above router.get function

Reply:
I'll test this and remove it if not needed.

Reply:
I've just removed the get method. Works perfectly fine!